src/excess_demand.jl

```diff
@@ -0,0 +1,24 @@
+function intermediate_goods_price_index(log_price_uk::Vector{T}, zOC::Vector{T},
+                                        tau::Vector{T}, mu::Vector{T}, gammaK::Vector{T},
+                                        K0::Vector{T}, xi::T) where {T <: Real}
+
+    # Computes the intermediate goods price index, Step 1 of ExcessDemand.m
+
+    pdYBar = Vector{T}(undef, length(log_price_uk))
+
+    for i in axes(log_price_uk, 1)
+
+        pdYBar[i] = intermediate_goods_price_index(log_price_uk[i], zOC[i], tau[i], mu[i], gammaK[i], K0[i], xi)
+
+    end
+
+    return pdYBar
```

Comment:
This can be done more simply with [comprehension](https://docs.julialang.org/en/v1/manual/arrays/#man-comprehensions):
```suggestion
    return [intermediate_goods_price_index(log_price_uk[i], zOC[i], tau[i], mu[i], gammaK[i], K0[i], xi) for i in eachindex(log_price_uk, zOC, ta, mu, gammaK, K0)]
```